pkg/distributor/distributor.go

```diff
@@ -154,6 +200,11 @@ func New(cfg Config, clientCfg client.Config, configs *runtime.TenantConfigs, in
 			Name:      "distributor_replication_factor",
 			Help:      "The configured replication factor.",
 		}),
+		streamSharding: promauto.With(registerer).NewCounter(prometheus.CounterOpts{
+			Namespace: "loki",
+			Name:      "distributor_stream_sharding",
```

Comment:
For counters, I'd recommend following the [convention](https://prometheus.io/docs/practices/naming/#metric-names) and adding a `_total` suffix.

```suggestion
			Name:      "distributor_streams_sharded_total",
```